### PR TITLE
Tentative fix for classloading issue in annotation processing

### PR DESCRIPTION
**Pull request by ljacomet**

### CODE REVIEW

**Comment by ljacomet** [comment]
@bot-gradle test RFM

**Comment by ljacomet** [comment]
@bot-gradle test RFM

**Comment by ljacomet** [comment]
@bot-gradle test RFM

**Comment by ljacomet** [comment]
@bot-gradle test RFM

File: subprojects/language-java/src/main/java/org/gradle/api/internal/tasks/compile/processing/AggregatingProcessingStrategy.java [hunk 1]
```diff
@@ -77,13 +80,48 @@ private void recordAggregatedTypes(Set<String> supportedAnnotationTypes, Set<? e
...
+        if (canAccessJDKTypes()) {
+            return filterElementsDirect(element);
+        } else {
+            return filterElementsReflection(element);
+        }
+    }
+
+    private static boolean filterElementsReflection(Element element) {
+        try {
+            Field sourceFile = element.getClass().getField("sourceFile");
```
Comment (melix) [inline subprojects/language-java/src/main/java/org/gradle/api/internal/tasks/compile/processing/AggregatingProcessingStrategy.java:98]: You could probably cache the `Field` per type.

**Comment by ljacomet** [comment]
@bot-gradle test RFM

**Review by melix — APPROVED** [review]

**Pull request merged**